grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -1004,17 +1004,17 @@ class StockMovementService {
             return shipment
         }
 
-        if (stockMovement.trackingNumber) {
-            ReferenceNumberType trackingNumberType = ReferenceNumberType.findById(Constants.TRACKING_NUMBER_TYPE_ID)
-            if (!trackingNumberType) {
-                throw new IllegalStateException("Must configure reference number type for Tracking Number with ID '${Constants.TRACKING_NUMBER_TYPE_ID}'")
-            }
+        ReferenceNumberType trackingNumberType = ReferenceNumberType.findById(Constants.TRACKING_NUMBER_TYPE_ID)
+        if (!trackingNumberType) {
+            throw new IllegalStateException("Must configure reference number type for Tracking Number with ID '${Constants.TRACKING_NUMBER_TYPE_ID}'")
+        }
 
-            // Needed to use ID since reference numbers is lazy loaded and equality operation was not working
-            ReferenceNumber referenceNumber = shipment.referenceNumbers.find { ReferenceNumber refNum ->
-                trackingNumberType?.id?.equals(refNum.referenceNumberType?.id)
-            }
+        // Needed to use ID since reference numbers is lazy loaded and equality operation was not working
+        ReferenceNumber referenceNumber = shipment.referenceNumbers.find { ReferenceNumber refNum ->
+            trackingNumberType?.id?.equals(refNum.referenceNumberType?.id)
+        }
 
+        if (stockMovement.trackingNumber) {
```

Comment:
Can you guys comment lines 1017-1032 to let me know what you're trying to accomplish (particularly with deleting the reference number)?

Reply:
@jmiranda they want to be able to remove value in trackingNumber field, so if there is no value in this field and there is a reference number then it's removed. Do you think we should set the referenceNumber.identifier to empty string instead deleting it?